### PR TITLE
Log in file every level of message

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
* Add everything in logs: messages, errors ...
* Add date & time before entries

```
[28-06-18 19:45:24] Starting upgrade...
[28-06-18 19:45:24] Shop deactivated. Now downloading... (this can take a while)
[28-06-18 19:45:24] Downloaded archive will come from https://download.prestashop.com/download/releases/prestashop_1.7.3.4.zip
[28-06-18 19:45:24] MD5 hash will be checked against ce0a59d7816e97d080a1fbb84e566a7b
[28-06-18 19:45:25] Downloading from https://download.prestashop.com/download/releases/prestashop_1.7.3.4.zip
[28-06-18 19:45:25] File will be saved in /var/www/html/admin-dev/autoupgrade/download/prestashop.zip
[28-06-18 19:45:25] Download directory has been emptied
```

### CODE REVIEW

File: classes/Log/LegacyLogger.php [hunk 1]
```diff
@@ -89,6 +89,10 @@ public function log($level, $message, array $context = array())
             return;
         }
 
+        if (!empty($this->fd)) {
+            fwrite($this->fd, '[' . date('d-m-y H:i:s') . '] ' . $message . PHP_EOL);
```
Comment (PierreRambaud) [inline classes/Log/LegacyLogger.php]: Better to write date like `Y-m-d H:i:s` for syslog and other logs parser tools

File: classes/Log/LegacyLogger.php [hunk 2]
```diff
@@ -89,6 +89,10 @@ public function log($level, $message, array $context = array())
             return;
         }
 
+        if (!empty($this->fd)) {
```
[the remark this answers is not in the document]
Comment (Quetzacoalt91) [reply classes/Log/LegacyLogger.php]: No, I had a case where fopen calls could return `false` or `null` (for instance when it was forbidden), that's why I chose this test.

File: classes/Log/LegacyLogger.php (hunk 1 above)
Comment (Quetzacoalt91) [reply classes/Log/LegacyLogger.php]: Here you go!

File: classes/Log/LegacyLogger.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (PierreRambaud) [reply classes/Log/LegacyLogger.php]: But is_resource on null or false will return false when fwrite / fopen need a resource :p

[the remark this answers is not in the document]
Comment (Quetzacoalt91) [reply classes/Log/LegacyLogger.php]: ... indeed :laughing:

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**